common/tier4_autoware_utils/src/vehicle/vehicle_state_checker.cpp

```diff
@@ -0,0 +1,123 @@
+// Copyright 2022 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "tier4_autoware_utils/vehicle/vehicle_state_checker.hpp"
+
+#include "tier4_autoware_utils/trajectory/trajectory.hpp"
+
+#include <string>
+
+namespace tier4_autoware_utils
+{
+VehicleStopChecker::VehicleStopChecker(rclcpp::Node * node)
+: clock_(node->get_clock()), logger_(node->get_logger())
+{
+  using std::placeholders::_1;
+
+  sub_odom_ = node->create_subscription<Odometry>(
+    "/localization/kinematic_state", rclcpp::QoS(1),
+    std::bind(&VehicleStopChecker::onOdom, this, _1));
+}
+
+bool VehicleStopChecker::isVehicleStopped(const double stop_duration = 0.0) const
+{
+  if (twist_buffer_.empty()) {
+    return false;
+  }
+
+  constexpr double stop_velocity = 1e-3;
+  const auto now = clock_->now();
+
+  const auto time_buffer_back = now - twist_buffer_.back().header.stamp;
+  if (time_buffer_back.seconds() < stop_duration) {
+    return false;
```

Comment:
If there is no older stamp velocity in buffer than `stop_duration`, it is impossible to check whether the vehicle is stopped more than `stop_duration`, so the function returns `false`.

Reply:
Sorry for the confusion. I mean, this check (time_buffer_back) is necessary but next one (time_buffer_front) can be removed.

Reply:
I see ! I'll remove it. Sorry for the confusion :pray:

Reply:
Fixed in https://github.com/autowarefoundation/autoware.universe/pull/896/commits/9a85dcfdfeac3c3d22721936966051aaf6a8168a.